pyseq.py

```diff
@@ -951,31 +955,39 @@ def uncompress(seq_string, fmt=global_format):
     except IndexError:
         pass
 
-    items = []
-    if missing:
-        for i in range(int(s), int(e) + 1):
-            if i in missing:
-                continue
-            f = pad % i
-            name = '%s%s%s' % (
-                match.groupdict().get('h', ''), f, 
-                match.groupdict().get('t', '')
-            )
-            items.append(Item(os.path.join(dirname, name)))
+    try:
+        head = match.group("h")
 
-    else:
-        for i in frames:
-            f = pad % i
-            name = '%s%s%s' % (
-                match.groupdict().get('h', ''), f, 
-                match.groupdict().get('t', '')
+    except IndexError:
+        head = ""
+
+    try:
+        tail = match.group("t")
+
+    except IndexError:
+        tail = ""
+
+    while len(missing) > 0:
+        m = missing.pop(0)
+        try:
+            frames.remove(m)
+        except IndexError:
+            pass
+
+    seq = None
+    for i in frames:
+        name = "%s%s%s" % (
+            head,
+            pad % i,
+            tail
             )
-            items.append(Item(os.path.join(dirname, name)))
+        item = Item(os.path.join(dirname, name))
+        if seq is None:
+            seq = Sequence([item])
+        else:
+            seq.append(item)
```

Comment:
With strict_pad=True (the default), this is throwing a SequenceError in the unit tests now

```
ERROR: test_uncompress_is_working_properly_7
...
    raise SequenceError('Item is not a member of this sequence')
SequenceError: Item is not a member of this sequence
```

All other tests pass. Previously, it was being ignored. Do we want uncompress throwing exceptions in cases like this? Just confirming, so I can update the unit test  before merging.

Reply:
it's been a while since I've looked at this.  But I think that would be correct.
However, this relates to the change that uncompress now returns a single Sequence or an empty list.  If uncompress should still handle a list of Sequences then it should probably create a new Sequence when an item doesn't match.  If uncompress does return a single sequence then we would assume that all Items created from the given string would match.

Reply:
Hmm, has uncompress ever handled multiple sequence strings? What would that look like? Quick test didn't work for me.

Reply:
Actually now that  i look at this it appears that the return type hasn't changed.  Not sure why I thought it had...
And yeah, I've never tried to uncompress multiple string sequences, and I don't think we would that in the standard uncompress.

Reply:
OK, great. So I'll update the unit tests and merge this in to the master. Thanks!